src/Arcus.WebApi.Tests.Unit/Security/Authorization/JwtTokenReaderTests.cs

```diff
@@ -77,5 +90,102 @@ public void Constructor_WithTokenValidationParametersAndDiscoveryEndpoint_Succee
             // Assert
             Assert.NotNull(jwtTokenReader);
         }
+
+        [Fact]
+        public void Constructor_WithClaimCheck_Succeeds()
+        {
+            // Arrange
+            var tokenValidationParameters = new TokenValidationParameters();
+            string openIdConnectDiscoveryUri = "https://login.microsoftonline.com/common/v2.0/.well-known/openid-configuration";
+
+            var claimCheck = new Dictionary<string, string>
+            {
+                {"aud", "368e1166-2235-48c7-8685-48d1fc24a9de"},
+                {"oid", "fa323e12-e4b8-4e22-bb2a-b18cb4b76301"},
+                {"aad", "ecc1e1c0-61bc-4e87-bd41-02333aa485bd"}
+            };
+
+            // Act
+            var jwtTokenReader = new JwtTokenReader(tokenValidationParameters, openIdConnectDiscoveryUri, claimCheck);
+
+            // Assert
+            Assert.NotNull(jwtTokenReader);
+        }
+
+        [Fact]
+        public async Task Constructor_ValidateClaimCheck_Succeeds()
```

Comment:
I would like to see some tests with more boundary test cases. Empty claim checks in both the configureed as the incoming, same identities but different claim check values, claim check with empty claim check values (should throw here), ... Now it's fairly limited as to what's being tested.